tests/lib/unit/API/Repository/Value/Filter/FilterTest.php

```diff
@@ -0,0 +1,323 @@
+    public function getInvalidSortClausesData(): iterable
+    {
+        yield [
+            [
+                new SortClause\Location\Priority(),
+                1,
+            ],
+            'Expected an instance of "eZ\Publish\API\Repository\Values\Content\Query\SortClause", ' .
+            'got "integer" at position 1',
+        ];
+
+        yield [
+            [
+                new SortClause\Location\Depth(),
+                new URLQuerySortClause\URL(Query::SORT_DESC),
+                Query::SORT_ASC,
+            ],
+            'Expected an instance of "eZ\Publish\API\Repository\Values\Content\Query\SortClause", ' .
+            'got "eZ\Publish\API\Repository\Values\URL\Query\SortClause\URL" at position 1',
+        ];
+
+        yield [
+            [
+                new SortClause\DatePublished(),
+                new SortClause\SectionIdentifier(Query::SORT_DESC),
+                Query::SORT_ASC,
+                new class('', Query::SORT_DESC) extends URLQuerySortClause {
+                },
+            ],
+            'Expected an instance of "eZ\Publish\API\Repository\Values\Content\Query\SortClause", ' .
+            'got "string" at position 2',
+        ];
+    }
+
+    /**
+     * @covers \Ibexa\Platform\API\Repository\Value\Filter\Filter::withCriterion
+     * @covers \Ibexa\Platform\API\Repository\Value\Filter\Filter::getCriterion
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException
+     */
+    public function testWithCriterion(): Filter
+    {
+        $filter = new Filter();
+        self::assertNull($filter->getCriterion());
+        $criterion = new Criterion\ContentId(1);
+        $filter->withCriterion($criterion);
+        self::assertEquals($criterion, $filter->getCriterion());
+
+        return $filter;
+    }
+
+    /**
+     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException
+     */
+    public function testWithCriterionThrowsBadStateException(): void
+    {
+        $filter = new Filter();
+        $filter->withCriterion(new Criterion\ParentLocationId(2));
+
+        $this->expectException(BadStateException::class);
+        $filter->withCriterion(new Criterion\ContentId(2));
+    }
+
+    /**
+     * @covers \Ibexa\Platform\API\Repository\Value\Filter\Filter::andWithCriterion
+     * @covers \Ibexa\Platform\API\Repository\Value\Filter\Filter::getCriterion
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException
+     */
+    public function testAndWithCriterion(): Filter
+    {
+        $criterion1 = new Criterion\ContentId(1);
+        $criterion2 = new Criterion\RemoteId(md5('/1/2/3/'));
+        $criterion3 = new Criterion\Ancestor('/1/2/');
+
+        $filter = new Filter();
+        $filter->withCriterion($criterion1);
+
+        $filter->andWithCriterion($criterion2);
+        $expectedCriterion = new Criterion\LogicalAnd([$criterion1, $criterion2]);
+        self::assertEquals($expectedCriterion, $filter->getCriterion());
```

Comment:
Hmmm AFAICS we rather use `$this->assertEquals(...)` (727 occurrences) syntax instead of `self::assertEquals(...)` (131 occurrences) in `\eZ\Publish\API\Repository\Tests\`...

Reply:
> Hmmm AFAICS we rather use `$this->assertEquals(...)` (727 occurrences) syntax instead of `self::assertEquals(...)` (131 occurrences) in `\eZ\Publish\API\Repository\Tests\`...

I believe calling a static method using `$this` is wrong, though indeed PHPUnit doc says "use whatever".